### PR TITLE
DiskInfo stats and Notification senders

### PR DESCRIPTION
**Pull request by jmendeza**
DiskInfo stats
Notification senders
https://github.com/craftercms/craftercms/issues/8272

## Summary by CodeRabbit

* **New Features**
  * Introduced disk usage monitoring capabilities, including retrieval of total, free, and used disk space with human-readable formatting.
  * Added support for sending notifications via email and webhooks, with configurable templates and payload handling.
  * Enhanced notification system with customizable message formatting and error handling.

* **Bug Fixes**
  * Improved validation for disk monitoring to handle non-existent paths and edge cases.

* **Tests**
  * Added unit tests for disk usage calculations, including scenarios with large disk sizes and invalid paths.

### CODE REVIEW

File: utilities/src/main/java/org/craftercms/commons/notification/impl/EmailSender.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	@Override
+	public void init(Configuration config) throws ConfigurationException {
+		super.init(config);
+		from = getStringProperty(config, FROM_CONFIG_KEY, from);
+		to = getRequiredStringArrayProperty(config, TO_CONFIG_KEY);
+		subject = getStringProperty(config, SUBJECT_CONFIG_KEY, subject);
+		html = getBooleanProperty(config, HTML_CONFIG_KEY, html);
+	}
```
[the remark this answers is not in the document]
Comment (jmendeza) [reply utilities/src/main/java/org/craftercms/commons/notification/impl/EmailSender.java:90]: A MessagingException would be thrown if `to` is not a valid address

File: utilities/src/main/java/org/craftercms/commons/notification/impl/EmailSender.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+		File tempFilePayload = null;
+		try {
+			File attachment = File.createTempFile("payload", ".json");
+			objectMapper.writeValue(attachment, payload);
+			tempFilePayload = attachment;
+		} catch (IOException e) {
+			logger.error("Failed to write payload of type {} to JSON", payload.getClass().getName(), e);
+		}
+		return tempFilePayload;
+	}
```
[the remark this answers is not in the document]
Comment (jmendeza) [reply utilities/src/main/java/org/craftercms/commons/notification/impl/EmailSender.java]: Updated

File: utilities/src/main/java/org/craftercms/commons/monitoring/DiskInfo.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		}
+		this.rootPath = rootFile.getPath();
+		totalSpace = rootFile.getTotalSpace();
+		freeSpace = rootFile.getFreeSpace();
+		usedSpace = totalSpace - freeSpace;
+		if (totalSpace == 0) {
+			throw new IllegalArgumentException("The provided path does not have a valid total space: " + rootPath);
+		}
+		diskUsage = (int) ((double) usedSpace / totalSpace * 100);
+	}
```
[the remark this answers is not in the document]
Comment (jmendeza) [reply utilities/src/main/java/org/craftercms/commons/monitoring/DiskInfo.java]: Updated

**Review by sumerjabri — APPROVED** [review]
